File: packages/core/src/memory/MemoryStore.ts
import { MemoryItem } from "./MemoryItem";

export interface MemoryStore {
  add(memory: MemoryItem): Promise<void>;
  retrieve(userId: string, query: string, limit?: number): Promise<MemoryItem[]>;
  list(userId: string): Promise<MemoryItem[]>;
  clear(userId: string): Promise<void>;
}

File: packages/compose/src/main/kotlin/ai/mobileagentos/compose/memory/MemoryStore.kt
package ai.mobileagentos.compose.memory

interface MemoryStore {
    suspend fun add(item: MemoryItem)
    suspend fun retrieve(userId: String, query: String, limit: Int = 5): List<MemoryItem>
    suspend fun list(userId: String): List<MemoryItem>
    suspend fun clear(userId: String)
}